### PR TITLE
Swagger CodeGen for WaaS [BA-5746]

### PR DESCRIPTION
**Pull request by kcibul**
Basic Swagger CodeGen support for a Java client for WaaS.  This uses the same approach as Sam to generate it's code.

Notes:
* Cromwell's entire YAML wasn't ready for swagger codegen, so the WaaS endpoint was placed into it's own 
* File -> String for the describe endpoint workflowSource parameter to make for a friendlier client.  Since this is only used for code generation that seems ok.  If this were to be driven off the same YAML as is served in a public swagger this would need more thought.  It might make sense to `sed s/File/string/` on the YAML before code generating.
* Future work would be to incorporate this into the travis build process, but I wanted to get this reviewed/merged first and we can tackle that as a separate PR.

### CODE REVIEW

File: codegen_java/build.sbt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      "com.squareup.okhttp3" % "logging-interceptor" % "3.12.1",
+      "com.google.code.gson" % "gson" % "2.8.5",
+      "org.apache.commons" % "commons-lang3" % "3.8.1",
+      "org.apache.oltu.oauth2" % "org.apache.oltu.oauth2.client" % "1.0.1",
+      "org.threeten" % "threetenbp" % "1.3.5" % "compile",
+      "io.gsonfire" % "gson-fire" % "1.8.0" % "compile",
+      "junit" % "junit" % "4.12" % "test",
+      "com.novocode" % "junit-interface" % "0.10" % "test"
+    )) ++ publishSettings:_*
+  )
```
Comment (aednichols) [inline codegen_java/build.sbt]: I do not personally care but I suspect folks will want newlines at end of file

File: codegen_java/project/Publishing.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import sbt.Keys._
+import sbt._
+import Artifactory._
+
+ /**
+  * NOTE: This was lifted wholesale from Cromwell.
```
Comment (aednichols) [inline codegen_java/project/Publishing.scala]: It was lifted from Cromwell for use in Rawls, but now that it's been re-lifted into Cromwell from Rawls this comment is out of date 😄 

Any chance we could DRY this out by re-using the original Cromwell edition?

File: codegen_java/build.sbt (hunk 1 above)
Comment (cjllanwarne) [reply codegen_java/build.sbt]: Yes please! 

I'll note that there's an intellij option to automatically add new-lines to files which is really helpful for this

**Comment by cjllanwarne** [comment]
I'm really not a big fan of copy/pasting huge chunks of the swagger doc over and checking it into the repo separately.

Is there an option in codegen to target only a subset of the endpoints? If not, is there a way to derive this cut-down swagger from the main one before running codegen over it? (eg programmatically parse the yaml, select only the womtool sections and re-write just that section to a file?)

**Comment by geoffjentry** [comment]
One thought is that the Cromwell swagger could be composed of multiple sub-swaggers: https://github.com/maxdome/swagger-combine

File: codegen_java/project/Publishing.scala (hunk 2 above)
Comment (kcibul) [reply codegen_java/project/Publishing.scala]: I took a look and Cromwell's Publishing does a ton more than this does.  My guess is this was lifted a long time ago.  I'll remove the comment though so it's not confusing.

**Comment by kcibul** [comment]
@geoffjentry @cjllanwarne I agree -- I poked at this for a while longer today and managed to get the original cromwell.yaml generating client code.  I put comments into the gen_java_client.sh as to how it happens

**Comment by kcibul** [comment]
> It would be nice to document how to run this - I'm guessing it's running one of the shell scripts?

@aednichols -- I had that thought, but it wasn't clear where they would go.  I see three options
1.) main readme
2.) separate readme (CLIENT.md)
3.) readthedocs site

Which is the appropriate path for docs like this?

**Comment by cjllanwarne** [comment]
IMO we should document this internally, since IMO the "forever home" for the process is to be wrapped up into the publish script for a new client version to be created every time we publish a new Cromwell version.

Until then, I'd put the "how to make clients" docs either in our releasing Cromwell google doc, or as another heading in https://github.com/broadinstitute/cromwell/tree/develop/processes/release_processes

**Comment by danbills** [comment]
this is great!  I'm working on how to merge this w/ our bigger SBT build to DRY it up

**Comment by kcibul** [comment]
@cjllanwarne -- I added to the releasing doc as you suggested.    Look good?

**Comment by kcibul** [comment]
@danbills how do you want to work your DRY?  Once I get a thumb from @cjllanwarne how about I merge and then you can have a separate PR so as not to link the two?

**Comment by cjllanwarne** [comment]
@kcibul just FYI, I've handed the review baton over to @danbills since he's way more aware of sbt syntax, etc than I am.

**Comment by danbills** [comment]
Once tests pass we can merge.  Will iterate on SBT if necessary, okay with me how it is separate

**Pull request merged**